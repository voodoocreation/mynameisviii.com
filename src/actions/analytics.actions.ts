import actionCreatorFactory from "typescript-fsa";

const createAction = actionCreatorFactory("ANALYTICS");

export const trackEvent = createAction<{
  event: string;
  [index: string]: any;
}>("TRACK_EVENT");
