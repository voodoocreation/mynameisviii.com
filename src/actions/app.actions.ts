import actionCreatorFactory from "typescript-fsa";

const createAction = actionCreatorFactory("APP");

export const initApp = createAction.async<{ locale?: string }, {}>("INIT");

export const toggleNavigation = createAction("TOGGLE_NAVIGATION");

export const setOnlineStatus = createAction<boolean>("SET_ONLINE_STATUS");

export const setHasNewVersion = createAction<boolean>("SET_HAS_NEW_VERSION");

export const setCurrentRoute = createAction<string>("SET_CURRENT_ROUTE");

export const changeRoute = createAction.async<string, {}>("CHANGE_ROUTE");
