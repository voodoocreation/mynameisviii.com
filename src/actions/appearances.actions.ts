import actionCreatorFactory from "typescript-fsa";

import { IAppearance, IDynamoResponse, ILatLng } from "../models/root.models";

const createAction = actionCreatorFactory("APPEARANCES");

export const fetchAppearances = createAction.async<
  {},
  IDynamoResponse<IAppearance, "startingAt">
>("FETCH");

export const fetchMoreAppearances = createAction.async<
  {},
  IDynamoResponse<IAppearance, "startingAt">
>("FETCH_MORE");

export const fetchAppearanceBySlug = createAction.async<string, IAppearance>(
  "FETCH_BY_SLUG"
);

export const setCurrentAppearanceSlug = createAction<string>(
  "SET_CURRENT_SLUG"
);

export const geocodeCurrentAppearanceAddress = createAction.async<{}, ILatLng>(
  "GEOCODE_CURRENT_ADDRESS"
);
