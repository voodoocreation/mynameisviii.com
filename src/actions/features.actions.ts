import actionCreatorFactory from "typescript-fsa";

const createAction = actionCreatorFactory("FEATURES");

export const addFeatures = createAction<string | string[]>("ADD");

export const removeFeatures = createAction<string | string[]>("REMOVE");
