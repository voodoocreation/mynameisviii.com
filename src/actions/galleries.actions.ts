import actionCreatorFactory from "typescript-fsa";

import { IGallery, IS3Response } from "../models/root.models";

const createAction = actionCreatorFactory("GALLERIES");

export const fetchGalleries = createAction.async<{}, IS3Response<IGallery>>(
  "FETCH"
);

export const fetchMoreGalleries = createAction.async<{}, IS3Response<IGallery>>(
  "FETCH_MORE"
);

export const fetchGalleryBySlug = createAction.async<string, IGallery>(
  "FETCH_BY_SLUG"
);

export const setCurrentGallerySlug = createAction<string>("SET_CURRENT_SLUG");
