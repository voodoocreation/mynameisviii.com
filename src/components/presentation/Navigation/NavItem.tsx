import * as React from "react";
import { FormattedMessage } from "react-intl";

import Link from "../Link/Link";

interface IProps {
  href?: string;
  isSelected: boolean;
  messageId: string;
  onClick?: () => void;
  params?: {};
  prefetch?: boolean;
  route?: string;
}

const NavItem: React.SFC<IProps> = ({
  isSelected,
  messageId,
  onClick,
  ...linkProps
}) =>
  isSelected ? (
    <li className="isSelected" onClick={onClick}>
      <FormattedMessage id={messageId} />
    </li>
  ) : (
    <li onClick={onClick}>
      <Link {...linkProps}>
        <FormattedMessage id={messageId} />
      </Link>
    </li>
  );

NavItem.defaultProps = {
  onClick: () => undefined
};

export default NavItem;
