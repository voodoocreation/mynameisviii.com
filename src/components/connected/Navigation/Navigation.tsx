import cn from "classnames";
import * as React from "react";
import { FormattedMessage, InjectedIntlProps, injectIntl } from "react-intl";
import { connect } from "react-redux";

import * as actions from "../../../actions/root.actions";
import { TStoreState } from "../../../reducers/root.reducers";
import * as selectors from "../../../selectors/root.selectors";
import Button from "../../presentation/Button/Button";
import NavItem from "../../presentation/NavItem/NavItem";

import "./Navigation.scss";

interface IProps extends InjectedIntlProps {
  currentRoute?: string;
  isOpen: boolean;
  setCurrentRoute: typeof actions.setCurrentRoute;
  toggleNavigation: typeof actions.toggleNavigation;
}

class Navigation extends React.Component<IProps> {
  public render() {
    return (
      <nav
        className={cn("Navigation", { isOpen: this.props.isOpen })}
        role="navigation"
      >
        <Button isStyled={false} onClick={this.onMenuButtonClick}>
          <FormattedMessage id="MENU" />
        </Button>

        <ul>
          {this.renderNavItem("/news", "NEWS_TITLE")}
          {this.renderNavItem("/releases", "RELEASES_TITLE")}
          {this.renderNavItem("/appearances", "APPEARANCES_TITLE")}
          {this.renderNavItem("/galleries", "GALLERIES_TITLE")}
          {this.renderNavItem("/stems", "STEMS_TITLE")}
          {this.renderNavItem("/resources", "RESOURCES_TITLE")}
        </ul>
      </nav>
    );
  }

  private onMenuButtonClick = () => {
    this.props.toggleNavigation();
  };

  private renderNavItem = (route: string, messageId: string) => (
    <NavItem route={route} isSelected={this.props.currentRoute === route}>
      <FormattedMessage id={messageId} />
    </NavItem>
  );
}

const mapState = (state: TStoreState) => ({
  currentRoute: selectors.getCurrentRoute(state),
  isOpen: selectors.isNavOpen(state)
});

export default injectIntl(
  connect(
    mapState,
    actions
  )(Navigation)
);
