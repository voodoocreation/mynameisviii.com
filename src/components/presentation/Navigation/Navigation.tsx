import cn from "classnames";
import { withRouter, WithRouterProps } from "next/router";
import * as React from "react";
import { FormattedMessage } from "react-intl";

import NavItem from "./NavItem";

interface IProps extends WithRouterProps {}

interface IState {
  currentRoute?: string;
  isOpen: boolean;
}

class Navigation extends React.Component<IProps, IState> {
  public readonly state = {
    currentRoute: undefined,
    isOpen: false
  };

  private currentRouteTimeout: undefined | NodeJS.Timer = undefined;

  public componentDidMount() {
    this.currentRouteTimeout = setTimeout(() => {
      this.setState({
        currentRoute: this.props.router.route
      });
    }, 1);
  }

  public componentWillUpdate(nextProps: IProps) {
    if (this.currentRouteTimeout) {
      clearTimeout(this.currentRouteTimeout);
    }

    this.currentRouteTimeout = setTimeout(() => {
      this.setState({
        currentRoute: nextProps.router.route
      });
    }, 1);
  }

  public componentWillUnmount() {
    if (this.currentRouteTimeout) {
      clearTimeout(this.currentRouteTimeout);
    }
  }

  public render() {
    return (
      <nav
        className={cn("Navigation", { isOpen: this.state.isOpen })}
        role="navigation"
      >
        <button onClick={this.onMenuButtonClick}>
          <FormattedMessage id="MENU" />
        </button>
        <ul>
          {this.renderNavItem("/news", "NEWS_TITLE")}
          {this.renderNavItem("/releases", "RELEASES_TITLE")}
          {this.renderNavItem("/appearances", "APPEARANCES_TITLE")}
        </ul>
      </nav>
    );
  }

  private onMenuButtonClick = () => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  };

  private onNavItemClick = () => {
    this.setState({
      isOpen: false
    });
  };

  private renderNavItem = (route: string, messageId: string) => (
    <NavItem
      route={route}
      messageId={messageId}
      isSelected={this.state.currentRoute === route}
      onClick={this.onNavItemClick}
    />
  );
}

export default withRouter<any>(Navigation);
