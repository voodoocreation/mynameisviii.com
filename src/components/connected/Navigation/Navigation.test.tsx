import * as actions from "../../../actions/root.actions";
import ComponentTester from "../../../utilities/ComponentTester";
import Navigation from "./Navigation";

const component = new ComponentTester(Navigation, true);

describe("[connected] <Navigation />", () => {
  describe("when interacting with the component", () => {
    const { wrapper } = component.mount();

    it("clicks the toggle button", () => {
      wrapper.find("button").simulate("click");
    });

    it("dispatches actions.toggleNavigation", () => {
      expect(
        component.reduxHistory.filter(actions.toggleNavigation.match)
      ).toHaveLength(1);
    });

    it("matches snapshot", () => {
      expect(wrapper.render()).toMatchSnapshot();
    });
  });
});
