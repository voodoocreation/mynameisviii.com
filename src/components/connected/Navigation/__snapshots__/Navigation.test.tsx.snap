// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[connected] <Navigation /> when interacting with the component matches snapshot 1`] = `
<nav
  class="Navigation isOpen"
  role="navigation"
>
  <button
    class="Button"
    type="button"
  >
    <span
      class="Button--content"
    >
      Menu
    </span>
  </button>
  <ul>
    <li
      class=""
    >
      <a
        href="/news"
      >
        News
      </a>
    </li>
    <li
      class=""
    >
      <a
        href="/releases"
      >
        Releases
      </a>
    </li>
    <li
      class=""
    >
      <a
        href="/appearances"
      >
        Appearances
      </a>
    </li>
    <li
      class=""
    >
      <a
        href="/galleries"
      >
        Galleries
      </a>
    </li>
    <li
      class=""
    >
      <a
        href="/stems"
      >
        Stems
      </a>
    </li>
    <li
      class=""
    >
      <a
        href="/resources"
      >
        Resources
      </a>
    </li>
  </ul>
</nav>
`;
